src/roswire/proxy/roslaunch/roslaunch.py

```diff
@@ -53,7 +53,9 @@ def read(self,
             If the given launch file could not be found in the package.
         """
         filename = self.locate(filename, package=package)
-        reader = LaunchFileReader(shell=self._shell, files=self._files)
+        reader = ROS1LaunchFileReader(app_instance=None,
```

Comment:
#408

Reply:
`app_instance=None` is a very bad smell here

Reply:
Fixed. For the moment, I've moved the app_instance down into the ROS2LaunchFileReader only.